### PR TITLE
ORC-293: [C++] Fix RleEncoderV1 for case when sizeof(long) < sizeof(int64_t)

### PR DESCRIPTION
**Pull request by rip-nsk**
and minor related changes.

### CODE REVIEW

File: c++/src/RLEv1.cc [hunk 1]
```diff
@@ -148,7 +148,7 @@ void RleEncoderV1::write(int64_t value) {
         numLiterals += 1;
       } else {
         numLiterals -= static_cast<int>(MINIMUM_REPEAT - 1);
-        long base = literals[numLiterals];
+        int64_t base = literals[numLiterals];
```
Comment (rip-nsk) [inline c++/src/RLEv1.cc]: 'auto' is more suitable, but it seems its usage is avoided here

File: c++/src/RLEv1.cc [hunk 2]
```diff
@@ -169,11 +169,11 @@ void RleEncoderV1::writeVslong(int64_t val) {
 
 void RleEncoderV1::writeVulong(int64_t val) {
   while (true) {
-    if ((val & ~0x7f) == 0) {
+    if ((val & ~BASE_128_MASK) == 0) {
```
Comment (wgtmac) [inline c++/src/RLEv1.cc]: I don't think we need change here as ~0x7f will be implicitly converted to int64_t.
Same for the below.

Comment (rip-nsk) [reply c++/src/RLEv1.cc]: It is added to match code in RleDecoderV1::readLong

File: c++/include/orc/Common.hh [hunk 3]
```diff
@@ -69,7 +69,7 @@ namespace orc {
     UNKNOWN_WRITER = INT32_MAX
   };
 
-  enum CompressionKind {
+  enum CompressionKind : std::int64_t {
```
Comment (xndai) [inline c++/include/orc/Common.hh]: The definition of CompressionKind_MAX already makes CompressionKind a int64 type. Enum base type is a C++ 11 feature and should at least be configurable through orc-config.hh

Comment (rip-nsk) [reply c++/include/orc/Common.hh]: What is the reason for these constants?

Comment (xndai) [reply c++/include/orc/Common.hh]: We can potentially get rid of this enum and use proto::CompressionKind instead. It can be done in a separate change.

**Comment by xndai** [comment]
+1

File: c++/include/orc/Common.hh (hunk 3 above)
Comment (rip-nsk) [reply c++/include/orc/Common.hh]: I mean that constants CompressionKind_MAX and WriterVersion_MAX are not used and can be removed.

**Pull request closed without merging**